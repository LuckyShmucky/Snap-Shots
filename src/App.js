
import './App.css';
import { useState, useEffect } from 'react';
import GitHubCard from './components/GitHubCard';


function App() {
 

const [gitHubName, setGitHubName] = useState('')
const [imageSrc, setImageSrc] = useState('')

useEffect( ()=>{ 
  fetch('https://api.github.com/users/LuckyShmucky')
  .then(res => res.json())
  .then(data =>{
    setGitHubName(data.name)
    setImageSrc(data.avatar_url)
  })

 }, [])



  return (
    <div className="App">
      <h2>{gitHubName}</h2>
      <GitHubCard imageSrc={imageSrc}/>
    </div>
  );
}

export default App;
