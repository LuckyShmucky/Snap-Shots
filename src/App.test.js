import { render, screen, waitFor } from '@testing-library/react';
import renderer, { act }  from "react-test-renderer"
// import { act } from 'react-dom/test-utils';
import App from './App';

beforeEach(() => {
  // sets everything back to initial state before each test
  fetch.resetMocks();
})




test("receives GitHub name", async () => {
  fetch.mockResponseOnce(JSON.stringify({name: 'Christian Luciano'}))
  render(<App />)
  const gitHubName = await waitFor(() => screen.getByTitle('name-header'))
  expect(gitHubName.textContent).toBe('Christian Luciano')
})



///This is not working anymore for some reason 
test('renders a snapshot for app component', () => {
 const tree = renderer.create(<App/>).toJSON()
 
 expect(tree).toMatchSnapshot()
});

// test("To see if user's name is in h2", async ()=>{
//   fetch.mockResponseOnce(JSON.stringify({name: 'Christian Luciano'}))
//   render(<App/>)
//   const name = await waitFor(() => screen.getByRole('heading', { level: 2 }))
//   expect(name).toHaveTextContent("Christian Luciano")
// })

