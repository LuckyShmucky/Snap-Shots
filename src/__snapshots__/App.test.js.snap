// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot for app component 1`] = `
<div
  className="App"
>
  <h2 />
  <div
    className="card"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <img
      alt="profile picture"
      className="card-img-top"
      role="cardImage"
      src=""
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        Card Title
      </div>
      <p
        className="card-text"
      >
        Some quick example text to build on the card title and make up the bulk of the card's content.
      </p>
    </div>
  </div>
</div>
`;
